tasks/main.yml

```diff
@@ -16,15 +20,21 @@
 
 - name: Download Caddy
   get_url: url=https://caddyserver.com/download/build?os=linux&arch=amd64&features={{ caddy_features }} dest={{ caddy_home }}/caddy.tar.gz force=yes
-  when: caddy_releases_cache.changed or caddy_features_cache.changed
-  register: caddy_binary_cache
+  when: caddy_update and (caddy_releases_cache.changed or caddy_features_cache.changed)
```

Comment:
Why did you add this check? `caddy_releases_cache` and `caddy_features_cache` already depend on `caddy_update`